lib/libutils/ext/include/confine_array_index.h

```diff
@@ -86,7 +86,7 @@ static inline size_t confine_array_index(size_t index, size_t size) {
 #ifdef __arm__
 static inline size_t confine_array_index(size_t index, size_t size)
 {
-	size_t safe_index = 0;
+	size_t safe_index = index;
```

Comment:
The name `safe_index` is  misleading, how about renaming it to something neutral like `ret_val` instead?

Reply:
ok